### ISSUE TITLE
Tree Entity 'adjacency-list' documentation example doesn't work

### ISSUE THREAD

**Issue by lioneltay** [typeorm/typeorm#2540]
**Issue type:**

[ ] question
[ ] bug report
[ ] feature request
[x] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

The example for the 'adjacency-list' tree entity documentation doesn't work (http://typeorm.io/#/tree-entities/adjacency-list).

```ts
import {Entity, Column, PrimaryGeneratedColumn, ManyToOne, OneToMany} from "typeorm";

@Entity()
export class Category {

    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string;

    @Column()
    description: string;

    @ManyToOne(type => Category, category => category.children)
    parent: Category;

    @OneToMany(type => Category, category => category.parent)
    children: Category[];
}

...

manager.getTreeRepository(Category).findTrees();
```

It gives the following error

**RepositoryNotTreeError: Repository of the "Category" class is not a TreeRepository. Try to apply @Tree decorator on your entity.**

Adding the Tree decorator results in another error

```ts
...
@Entity()
@Tree("adjacency-list")
export class Category {
...
```

**Cannot read property 'propertyName' of undefined.**

Adding TreeChildren and TreeParent results in another error

```ts
@Entity()
@Tree("adjacency-list")
export class Category {
  @PrimaryGeneratedColumn() id: number

  @Column() name: string

  @ManyToOne(type => Category, category => category.children)
  @TreeParent()
  parent: Category

  @OneToMany(type => Category, category => category.parent)
  @TreeChildren()
  children: Category[]
}
```

**Error: Supported only in tree entities**

**Comment by pleerock** [comment]
Documentation example is correct. `adjacency-list` is a primitive one, you can't use `TreeRepository` with it.

**Comment by lioneltay** [comment]
Oh I see, did not realise that. Is there a technical reason why the 'adjacency-list' cannot be used with the TreeRepository API?

Perhaps a note could be made in the docs or 'adjacency-list' could be removed from the union type of @Tree().

**Comment by pleerock** [comment]
> Is there a technical reason why the 'adjacency-list' cannot be used with the TreeRepository API?

yes adjacency-list is very primitive, and its basically the same as self-referencing relation. Its limited and can work only with a simple trees. You'll never be able to get all the tree data with this pattern.

> 'adjacency-list' could be removed from the union type of @Tree().

probably you are right, we should remove it

**Comment by rezolan** [comment]
How can i make where condition in this method this.pageRepository.findTrees()?
I want to get all levels of tree where in column isDeleted = false

**Comment by fullofcaffeine** [comment]
> yes adjacency-list is very primitive, and its basically the same as self-referencing relation. Its limited and can work only with a simple trees. You'll never be able to get all the tree data with this pattern.

That's a bold statement. Why couldn't we use the technique described [here](https://schinckel.net/2014/09/13/long-live-adjacency-lists/) (an article that's itself linked in the TypeORM tree docs) to get the whole tree?

**Comment by ghaedi1993** [comment]
Docs are misleading there should probably a line noting that TreeRepository is not going to work with as said primitive Adjacency list.

**Comment by RamenSayami** [comment]
> How can i make where condition in this method this.pageRepository.findTrees()? I want to get all levels of tree where in column isDeleted = false

i want to do the same. did u figure out how?
